Python/Product/PythonTools/PythonToolsPackage.cs

```diff
@@ -299,14 +299,26 @@ internal sealed class PythonToolsPackage : CommonPackage, IVsComponentSelectorPr
 
                 if (saveDirtyFiles) {
                     var rdt = provider.GetService(typeof(SVsRunningDocumentTable)) as IVsRunningDocumentTable;
-                    if (rdt != null) {
+                    var rdt4 = provider.GetService(typeof(SVsRunningDocumentTable)) as IVsRunningDocumentTable4;
+                    if (rdt != null && rdt4 != null) {
```

Comment:
Wait, so `GetDocumentMoniker(GetDocumentCookie(fileName)) != fileName` ?

Reply:
There's a save dirty documents in between, and that's what is changing the filename. In that bug, the filename is in %temp% folder, and after the save dirty, it's wherever the user saved it.